pipeline/pipeline-notes.rst

```diff
@@ -14,47 +14,98 @@ and will then compile the paper from scratch using the new figures.
 Ubuntu 14.04 instance. The c3.2xlarge instance size will work just fine, but
 do not choose a smaller instance than this.
 
 Make sure you edit your security groups to include port 22 (SSH) and port 
-80 (HTTP) ; you'll need the first one to log in, and the second one to 
+8080 ; you'll need the first one to log in, and the second one to 
 connect to the ipython notebook.
 
-Once you ssh in, set yourself up to run as root.::
+If you need to create a new user, you can do so with::
 
- sudo su
+ useradd -m -d /home/user <your_user_name>
+
+We will also want to link /mnt/bin directory to our home/bin directory::
+ ln -s /mnt/bin ${HOME}/bin
+
+
+RackSpace
+____________________________________________
+Additionally, if you are on RackSpace, you will need to disable the firewall so that you can run the iPython notebook.::
+
+  sudo service ufw stop
+
+If you need to create a new user, you can do so with::
+
+ adduser <your_user_name>
+
+
+Depending on the instance you chose, we may need to add extra space for data by formating and mounting a drive.
+Formating/Mounting Drive::
+
+  mkfs -t ext4 /dev/xvde1
+  mount /dev/xvde1 /home/<your_user_name> -t ext4
 
-Next, we are going to set the instance up with many of the software 
-packages we will need::
 
 
- apt-get update
- apt-get --yes install screen git curl gcc make g++ python-dev unzip \
+
+Root Installs
+--------------------------------------------
+Next, we are going to set the instance up with many of the software 
+packages we will need. You will need root permissions to install these::
+
+ sudo apt-get update
+ sudo apt-get --yes install screen git curl gcc make g++ python2.7-dev unzip \
             default-jre pkg-config libncurses5-dev r-base-core \
             r-cran-gplots python-matplotlib sysstat bowtie \
             texlive-latex-recommended mummer python-pip ipython \
-            ipython-notebook bioperl ncbi-blast+
+            ipython-notebook bioperl ncbi-blast+ python-virtualenv hmmer \
+            ncbi-tools-bin infernal aragorn parallel python-numpy
 
 
-Now, you'll need to install the version of 'khmer' that the
+
+Now you'll need to install the version of 'khmer' that the
 paper is currently using.::
- 
+
+ cd ${HOME}
+ mkdir -p bin
+ virtualenv venv
+ source venv/bin/activate
  easy_install -U setuptools
  pip install khmer==1.1
+ echo 'export PATH=${PATH}:${HOME}/bin' >> ${HOME}/.bashrc
+ source ${HOME}/.bashrc
 
-and Velvet. (We need to do this the old fashioned way to enable large k-mer
+
+All Non-Root
+--------------------------------------------
+Once you have completed all of the previous commands that require root permissions,
+go ahead and login as you normally would.
+Now we can go ahead and add ``~/bin`` to our path::
+
+ cd ${HOME}
+ source venv/bin/activate
+
+
+We need to install Velvet. (We need to do this the old fashioned way to enable large k-mer
 sizes)::
 
- cd /root
+ mkdir -p ${HOME}/src/velvet
```

Comment:
I know we solved this before, but I'm having a hard time granting non-root the right permissions:

`miguel@cloud-server-02:~/src$ mkdir -p ${HOME}/src/velvet`
`mkdir: cannot create directory ‘/home/miguel/src/velvet’: Permission denied`

 and these also require permissions:

    echo 'export PATH=${PATH}:${HOME}/bin' >> ${HOME}/.bashrc
    source ${HOME}/.bashrc



an `ls -la` on home/non-root yeilds:

drwxr-xr-x 6 root root  4096 Mar 17 13:44 .
drwxr-xr-x 7 root root  4096 Mar 17 13:08 ..
drwxr-xr-x 2 root root  4096 Mar 17 13:43 bin
drwx------ 2 root root 16384 Mar 17 13:41 lost+found
drwxr-xr-x 2 root root  4096 Mar 17 13:44 src
drwxr-xr-x 6 root root  4096 Mar 17 13:43 venv

I ran `chmod -R u+rw /home/non-root`, shouldn't that cover this?